lib/findExports.js

```diff
@@ -0,0 +1,203 @@
+      right.arguments[0].type === 'StringLiteral') {
+      // module.exports = require('someOtherFile.js');
+      const pathToRequiredFile = path.resolve(path.dirname(absPathToFile),
+        right.arguments[0].value);
+      const requiredFileContent = fs.readFileSync(
+      requireResolve(pathToRequiredFile), 'utf8');
+      // eslint-disable-next-line no-use-before-define
+      return findNamedExports(parse(requiredFileContent).program.body,
+        pathToRequiredFile);
+    }
+    // module.exports = { foo: 'foo' };
+    if (right.type === 'ObjectExpression') {
+      return right.properties.map(({ key }) => key.name).filter(Boolean);
+    }
+    if (right.type === 'Identifier') {
+      return definedNames[right.name];
+    }
+  }
+
+  if (!left.object || !left.property) {
+    return [];
+  }
+
+  if (left.object.type === 'MemberExpression' &&
+    left.object.object.name === 'module' &&
+    left.object.property.name === 'exports') {
+    // module.exports.foo = 'bar';
+    return [left.property.name];
+  }
+
+  if (left.type === 'MemberExpression' &&
+    left.object.type === 'Identifier' &&
+    left.object.name === 'exports') {
+    // exports.foo = 'bar';
+    return [left.property.name];
+  }
+
+  return [];
+}
+
+function findDefinedNames(node, definedNames) {
+  if (node.type === 'ExpressionStatement') {
+    const { left, right } = node.expression;
+    if (left && right) {
+      if (left.object) {
+        (definedNames[left.object.name] || []).push(left.property.name);
+      }
+    }
+  }
+  if (node.type !== 'VariableDeclaration') {
+    return;
+  }
+  node.declarations.forEach(({ id, init }) => {
+    if (!init) {
+      return;
+    }
+    if (init.type === 'ObjectExpression') {
+      definedNames[id.name] = // eslint-disable-line no-param-reassign
+        init.properties.map(({ key }) => key.name).filter(Boolean);
+    } else if (init.type === 'FunctionExpression') {
+      definedNames[id.name] = []; // eslint-disable-line no-param-reassign
+    }
+  });
+}
+
+function findNamedExports(nodes, absPathToFile, definedNames) {
+  const result = [];
+  nodes.forEach((node) => {
+    result.push(...findESNamedExports(node));
+    result.push(...findCommonJSExports(node, definedNames,
+      absPathToFile));
+  });
+  return result;
+}
+
+function hasDefaultExport(nodes) {
+  return nodes.some((node) => {
+    if (node.type === 'ExportDefaultDeclaration') {
+      return true;
+    }
+    if (node.type !== 'ExpressionStatement') {
+      return false;
+    }
+    // Potential CommonJS export
+    const { left, right } = node.expression;
+    if (!left || !right) {
+      return false;
+    }
+    if (left.name === 'exports') {
+      return true;
+    }
+    if (!left.object || !left.property) {
+      // foo = 'bar';
+      return false;
+    }
+    return left.object.name === 'module' &&
+      left.property.name === 'exports';
+  });
+}
+
+const DEFAULT_EXPORT_PATTERN = /\smodule\.exports\s*=\s*(\w+)/;
```

Comment:
If you start this pattern with `\s`, will it match if `module.exports` is on the first line without any whitespace before it? You might need to tweak this pattern.